packages/ember-glimmer/tests/integration/components/append-test.js

```diff
@@ -35,13 +35,235 @@ class AbstractAppendTest extends RenderingTest {
     this.ids.push(component.elementId);
   }
 
+  ['@test such lifecycle'](assert) {
+    let hooks = [];
+
+    let oldRegisterComponent = this.registerComponent;
+    let componentsByName = {};
+
+    // TODO: refactor/combine with other life-cycle tests
+    this.registerComponent = function(name, _options) {
+      function pushHook(hookName) {
+        hooks.push([name, hookName]);
+      }
+
+      let options = {
+        ComponentClass: _options.ComponentClass.extend({
+          init() {
+            expectDeprecation(() => { this._super(...arguments); }, /didInitAttrs called/);
+            if (name in componentsByName) {
+              throw new TypeError('Component named: ` ' + name + ' ` already registered');
+            }
+            componentsByName[name] = this;
+            pushHook('init');
+            this.on('init', () => pushHook('on(init)'));
+          },
+
+          didInitAttrs(options) {
+            pushHook('didInitAttrs', options);
+          },
+
+          didReceiveAttrs() {
+            pushHook('didReceiveAttrs');
+          },
+
+          willInsertElement() {
+            pushHook('willInsertElement');
+          },
+
+          willRender() {
+            pushHook('willRender');
+          },
+
+          didInsertElement() {
+            pushHook('didInsertElement');
```

Comment:
We should do the same assertions as the other lifecycle tests. Things like element is present (can't guarantee it's in DOM though since we don't know where the element we were told is), current state, has parent view or not, etc